modules/xfem/src/base/XFEM.C

```diff
@@ -967,6 +989,31 @@ XFEM::cutMeshWithEFA()
 
       libmesh_elem->set_node(j) = libmesh_node;
 
+      // Store solution for all nodes affected by XFEM (even existing nodes)
+      if (parent_elem->is_semilocal(_mesh->processor_id()))
+      {
+        Node * solution_node = libmesh_node; //Node from which to store solution
+
+        for (auto nn : NewNodes)
+        {
+          if (nn->id() == libmesh_node->id())
+          {
+            solution_node = _mesh->node_ptr(nn->parent()->id());
```

Comment:
Just so it's on your radar. This line of code will fail badly if we attempt to run with DistributedMesh. I know that's not your goal here but at least get in the habit of using mooseAssert so that when we try we don't have to dig in with the debugger to find all these problems.